lib/si-db/src/transactions.rs

```diff
@@ -0,0 +1,21 @@
+use si_data_nats::NatsTxn;
+use si_data_pg::{
+    PgError,
+    PgTxn,
+};
+use thiserror::Error;
+
+pub trait SiDbTransactions {
+    fn pg(&self) -> &PgTxn;
+    fn nats(&self) -> &NatsTxn;
+}
+
+// TODO TransactionsError really needs to be moved to a common place accessible here and in dal
+#[remain::sorted]
+#[derive(Debug, Error, strum::EnumDiscriminants)]
+pub enum SiDbTransactionsError {
```

Comment:
Why keep this alive?

Reply:
Mainly because one thing at a time, honestly :) I think the right thing to do in the end is to call this `TransactionsError` and remove dal::TransactionsError, because most of the things returning that error aren't actually transactions ... but that will be better done by itself.

Reply:
Sounds great to me!